frontend/src/components/pagedItemsTable/serializeFilterParamsTS.ts

```diff
@@ -4,128 +4,147 @@
+                    else if (options.startsWith) key += '__startswith'
+                    else if (options.exactMatch) {
+                        /* Left blank because exact not uniformly accepted by the backend */
+                    }
+                    else key += '__icontains'
+                } else {
+                    key += '__icontains'
+                }
+
+                params[key] = value
+            } else {
+                console.error('Invalid INPUT filter value', value)
             }
-            else key += '__icontains'
-          } else {
-            key += '__icontains'
-          }
-
-          params[key] = value
+            break
         }
-        break
-      }
-
-      case FILTER_TYPE.INPUT_NUMBER:
-      case FILTER_TYPE.INPUT_OBJECT_ID: {
-        if (value) {
-          key += '__in'
-          params[key] = value
+
+        case FILTER_TYPE.INPUT_NUMBER:
+        case FILTER_TYPE.INPUT_OBJECT_ID: {
+            if (value) {
+                key += '__in'
+                params[key] = value
+            } else {
+                console.error('Invalid INPUT_NUMBER/INPUT_OBJECT_ID filter value', value)
+            }
+            break
         }
-        break
-      }
-
-      case FILTER_TYPE.METADATA: {
-        if (isMetadataFilterValue(value)) {
-          //Serialize key-value metadata pairs in a string
-          //with the form: name1__value1, name2__value2, name2__value3, etc
-          params[key] = (value as MetadataFilterValue).reduce((serializedMetadata, metadata) => {
-            return serializedMetadata + metadata.name + '__' + (metadata.value ? metadata.value : '') + ','
-          }, '')
+
+        case FILTER_TYPE.METADATA: {
+            if (isMetadataFilterValue(value)) {
+                //Serialize key-value metadata pairs in a string
+                //with the form: name1__value1, name2__value2, name2__value3, etc
+                params[key] = (value as MetadataFilterValue).reduce((serializedMetadata, metadata) => {
+                    return serializedMetadata + metadata.name + '__' + (metadata.value ? metadata.value : '') + ','
+                }, '')
+            } else {
+                console.error('Invalid METADATA filter value', value)
+            }
+            break
         }
-        break
-      }
 
-      default: {
-        throw new Error('Invalid filter type')
-      }
+        default: {
+            throw new Error('Invalid filter type')
+        }
     }
-  })
 
-  return params
+    return params
+}
+
+export default function serializeFilterParamsWithDescriptions(filters: FilterSet) {
+    const params: NonNullable<QueryParams> = {}
+
+    Object.keys(filters).forEach((key) => {
+        const setting = filters[key]
+        if (setting.value && setting.description) {
+            const paramsForFilterKey = paramsForFilterKeyAndSetting(key, setting.value, setting.description, setting.options)
```

Comment:
Not sure why you made a function out of this ... will it ever be called anywhere but here ? Just forces us to scroll around the file to find the rest of the code ...

Reply:
I factored it out in case we want to obtain query parameters for one particular column filter-options. It's a left-over relic from my work in another branch.